File: docs/src/test/java/org/springframework/security/docs/features/authentication/authenticationpasswordstoragescrypt/SCryptPasswordEncoderUsage.java
package org.springframework.security.docs.features.authentication.authenticationpasswordstoragescrypt;

import static org.junit.Assert.assertTrue;

import org.springframework.security.crypto.scrypt.SCryptPasswordEncoder;

public class SCryptPasswordEncoderUsage {
	void testSCryptPasswordEncoder() {
		// tag::sCryptPasswordEncoder[]
		// Create an encoder with all the defaults
		SCryptPasswordEncoder encoder = SCryptPasswordEncoder.defaultsForSpringSecurity_v5_8();
		String result = encoder.encode("myPassword");
		assertTrue(encoder.matches("myPassword", result));
		// end::sCryptPasswordEncoder[]
	}
}

File: docs/src/test/kotlin/org/springframework/security/kt/docs/features/authentication/authenticationpasswordstoragescrypt/SCryptPasswordEncoderUsage.kt
package org.springframework.security.kt.docs.features.authentication.authenticationpasswordstoragescrypt

import org.junit.Assert.assertTrue
import org.springframework.security.crypto.scrypt.SCryptPasswordEncoder

class SCryptPasswordEncoderUsage {
    fun testSCryptPasswordEncoder() {
        // tag::sCryptPasswordEncoder[]
        // Create an encoder with all the defaults
        val encoder = SCryptPasswordEncoder.defaultsForSpringSecurity_v5_8()
        val result: String? = encoder.encode("myPassword")
        assertTrue(encoder.matches("myPassword", result))
        // end::sCryptPasswordEncoder[]
    }
}
